lib/api/confighandler.go

```diff
@@ -0,0 +1,369 @@
+// Copyright (C) 2014 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package api
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"golang.org/x/crypto/bcrypt"
+
+	"github.com/syncthing/syncthing/lib/config"
+	"github.com/syncthing/syncthing/lib/protocol"
+	"github.com/syncthing/syncthing/lib/sync"
+)
+
+const configBase = "/rest/config/"
```

Comment:
This feels like it should be a parameter to newConfigHandler, like it is to the other HandleFoo methods that register handlers for things?

Reply:
Having the "legacy" paths hardcoded in there isn't super pretty either. How about exposing at least the legacy operations as handler funcs on the configHandler and registering them like normal in the api service?